vllm/engine/arg_utils.py

```diff
@@ -364,7 +363,6 @@ class EngineArgs:
     max_prompt_adapter_token: int = \
         PromptAdapterConfig.max_prompt_adapter_token
 
-    device: Device = DeviceConfig.device
```

Comment:
Are we sure this is not longer supported?
Now the upstream caller still pass in the param, and will cause an error
```bash
lm_eval   --model vllm   --model_args "pretrained=Qwen/Qwen3-30B-A3B-FP8,max_model_len=32768,enforce_eager=True"   --trust_remote_code   --tasks gsm8k   --num_fewshot 5   --batch_size auto
```

Reply:
Also, if we are removing this, should we also have removed the Device config from config.py?

Reply:
@yewentao256 See https://github.com/vllm-project/vllm/issues/18301#issuecomment-2896683354, the `device` parameter will not take effect even if it is set. It is a useless parameter. If you want to specify the device, you can specify it through the `CUDA_VISIABLE_DEVICES` environment variable. If it is cpu mode, you need to recompile it to cpu mode.

Reply:
@hmellor Thanks for your reminder, I will take a closer look at how to safely remove `DeviceConfig.device`, which requires another PR.

Reply:
> @yewentao256 See [#18301 (comment)](https://github.com/vllm-project/vllm/issues/18301#issuecomment-2896683354), the `device` parameter will not take effect even if it is set. It is a useless parameter. If you want to specify the device, you can specify it through the `CUDA_VISIABLE_DEVICES` environment variable. If it is cpu mode, you need to recompile it to cpu mode.
Make sense, thanks!